Cost Tracking Change Log (#347)
vellum-ai/vellum-client-generator: fern/docs/content/change-log/2024-08.mdx

title: Changelog | August, 2024
---

> ## Prompt Cost Tracking

Reviewer: ```suggestion
## Prompt Sandbox Cost Tracking
```

> _August 9th, 22024_

Reviewer 2: ```suggestion
_August 9th, 2024_
```

> You can now see cost tracking for input and output tokens in prompt sandboxes, both in the prompt editor and in the comparison tab!

Reviewer: ```suggestion
You can now see the dollar cost of a Prompt's execution within both a Prompt Sandbox's Prompt Editor and Comparison Mode views.
```

> Cost of input and output tokens are being tracked from providers.

Reviewer: These costs are calculated using model providers' publicly available pricing data in conjunction with the number of input/output tokens used.

![Prompt Sandbox With Cost Tracking](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-08/prompt_sandbox_with_cost_tracking.png)

> ![Prompt Sandbox Comparison With Cost Tracking](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-08/prompt_sandbox_comparison_with_cost_tracking.png)

Reviewer: Looking at these images, I feel like we should round to 9 significant digits. the long-tail of 99999999 feels overkill.

Reviewer: Also, while true, the definition of Vellum here is pretty gruesome and makes me think we could use a screenshot of another example 😅

> If you're curious about the pricing information, feel free to look into a Model's details page.
> ![MLModel Detail Page with Billing Config](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-08/mlmodel_detail_page_with_cost_config.png)
> 
> Most popular models are already supported, with even more models following in the coming days.

Reviewer: ```suggestion
If you're curious about a given model's pricing, you can view details in the Model's detail page.
![MLModel Detail Page with Billing Config](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-08/mlmodel_detail_page_with_cost_config.png)

Most popular models already have pricing information populated, with support for even more models following in the coming days.

Showing cost information in Prompt Sandboxes is just the first step! We'll expose cost details throughout more of Vellum over time.
```

This is particularly helpful if you want to do things like compare the results of two different Evaluation runs,
download the results of a past Evaluation run, or simply view the Test Cases that existed at that time.

> ## Prompt Cost Tracking

Reviewer 2: [blocking] let's add this towards the top, we go from most recent -> least recent

_August 8th, 22024_

> You can now see cost tracking in prompt sandboxes, both in the prompt editor and in the comparison tab!

Reviewer 2: [nit] Let's add some copy here that says that we are tracking the cost from the provider. Let's also add something towards the end hinting that it's available for OpenAI, Anthropic, and Google models, with the rest to follow in the coming days

![Prompt Sandbox With Cost Tracking](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-08/prompt_sandbox_with_cost_tracking.png)
![Prompt Sandbox Comparison With Cost Tracking](https://storage.cloud.google.com/vellum-public/help-docs/changelogs/2024-08/prompt_sandbox_comparison_with_cost_tracking.png)

> If you're curious about the pricing information, feel free to additionally look into a Model's page.

Reviewer 2: ```suggestion
If you're curious about the pricing information, feel free to look into a Model's details page.
```

> ![MLModel Detail Page with Billing Config](mlmodel_detail_page_with_cost_config.png)

Reviewer 2: [blocking] CI is breaking on this link
